executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -71,35 +91,46 @@ fn create_index<T: FieldElement>(
         });
 
     // create index for this lookup
+    let variable_name = |v: &FixedColOrConstant<T, PolyID>| match v {
+        FixedColOrConstant::FixedCol(poly_id) => fixed_data.column_name(poly_id).to_string(),
+        FixedColOrConstant::Constant(c) => c.to_string(),
+    };
     log::trace!(
         "Generating index for lookup in columns (in: {}, out: {})",
-        input_fixed_columns
-            .iter()
-            .map(|c| fixed_data.column_name(c).to_string())
-            .join(", "),
-        output_fixed_columns
-            .iter()
-            .map(|c| fixed_data.column_name(c).to_string())
-            .join(", ")
+        input_fixed_columns.iter().map(variable_name).join(", "),
+        output_fixed_columns.iter().map(variable_name).join(", ")
     );
 
     let start = std::time::Instant::now();
 
     // get all values for the columns to be indexed
     let input_column_values = input_fixed_columns
-        .iter()
-        .map(|id| fixed_data.fixed_cols[id].values_max_size())
+        .into_iter()
+        .map(|id| match id {
+            FixedColOrConstant::FixedCol(poly_id) => {
+                FixedColOrConstant::FixedCol(fixed_data.fixed_cols[&poly_id].values_max_size())
+            }
+            FixedColOrConstant::Constant(c) => FixedColOrConstant::Constant(c),
+        })
         .collect::<Vec<_>>();
 
     let output_column_values = output_fixed_columns
-        .iter()
-        .map(|id| fixed_data.fixed_cols[id].values_max_size())
+        .into_iter()
+        .map(|id| match id {
+            FixedColOrConstant::FixedCol(poly_id) => {
+                FixedColOrConstant::FixedCol(fixed_data.fixed_cols[&poly_id].values_max_size())
+            }
+            FixedColOrConstant::Constant(c) => FixedColOrConstant::Constant(c),
+        })
         .collect::<Vec<_>>();
 
     let degree = input_column_values
         .iter()
         .chain(output_column_values.iter())
-        .map(|values| values.len())
+        .filter_map(|p| match p {
+            FixedColOrConstant::FixedCol(values) => Some(values.len()),
+            FixedColOrConstant::Constant(_) => None,
```

Comment:
What if all of them are constants?

Reply:
Then it fails 😢 Doesn't seem to happen in practice though.

We have issues with these types of lookups already in other places, because if there is no column reference, we can't know the machine, and therefor can't know the size. This would be fixed if an identity (in this case the bus receive) strictly belonged to a namespace and we can reference it from the identity. But AFAIK, right now, identities can live anywhere and reference columns from any namespace.